File: amazia-core/src/main/java/com/example/inbound/controller/ListInboundController.java
package com.example.inbound.controller;

import com.example.inbound.dto.InboundResponse;
import com.example.inbound.service.ListInboundService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 入荷一覧 API（GET /api/inbounds[?productId=N]）。
 */
@RestController
@RequestMapping("/api")
public class ListInboundController {

    private final ListInboundService service;

    public ListInboundController(ListInboundService service) {
        this.service = service;
    }

    @GetMapping("/inbounds")
    public ResponseEntity<List<InboundResponse>> list(
            @RequestParam(required = false) Long productId) {
        return ResponseEntity.ok(service.list(productId));
    }
}

File: amazia-console/app/Inbound/Controller/ListInboundController.php
<?php

namespace App\Inbound\Controller;

use App\Http\Controllers\Controller;
use App\Inbound\Service\ListInboundService;
use Illuminate\Http\Request;

/**
 * 入荷一覧 Controller（GET /api/inbounds[?productId=N]）。
 */
class ListInboundController extends Controller
{
    private ListInboundService $service;

    public function __construct(ListInboundService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $productId = $request->query('productId');
        $response = $this->service->list($productId !== null ? (int) $productId : null);
        return response()->json($response->json(), $response->status());
    }
}
